crates/ra_syntax/src/parser_impl/event.rs

```diff
@@ -251,15 +251,26 @@ fn n_attached_trivias<'a>(
     match kind {
         CONST_DEF | TYPE_DEF | STRUCT_DEF | ENUM_DEF | FN_DEF | TRAIT_DEF | MODULE => {
             let mut res = 0;
+            // Whether we have stopped and are assumming that all later trivia is floating
+            let mut stopped = false;
             for (i, (kind, text)) in trivias.enumerate() {
                 match kind {
                     WHITESPACE => {
-                        if text.contains("\n\n") {
-                            break;
+                        if text.matches("\n").count() >= 2 && !stopped {
```

Comment:
FYI `matches` is disjoint so I think it will return 2 in the case of "\n\t\n".

Reply:
That's why I wanted it - it seemed to make no sense that in
```rust
// This is a comment
 
/// This is also a comment
struct Foo{}
```
the entire file was is within the STRUCT, but in
```rust
// This is a comment

/// This is also a comment
struct Foo{}
```
the first comment is not